### PR TITLE
Add C++ example for executing filters in place

### PR DESCRIPTION
**Pull request by blowekamp**

### CODE REVIEW

File: Examples/CppInPlace/Documentation.rst [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  Image SmoothingRecursiveGaussian(Image &&image1, std::vector<double> sigma, bool normalizeAcrossScale);
+
+Additionally, most C++ overloaded Image numeric operators such as `+`, `-`, `*`, `%`, etc. support inplace execution.
+
+The SimpleITK Image class supports move semantics with the constructor and assignment operator. Since the
+Image class also uses a lazy copy on write implement for the constructors and assignment there is not a  significant
+direct performance benefit using these operations. However, the move semantics can be useful to reduce the number
+of Image objects referencing the same image buffer to prevent unnecessary copies. Lastly, if a rvalue Image object is
+passed to a filter and Image object's buffer is not unique, then the image buffer will be copied before the filter is
+executed inplace. Generally, coping the image then executing the filter inplace is faster than executing the non-inplace
```
Comment (zivy) [inline Examples/CppInPlace/Documentation.rst:36]: coping -> copying

**Review by zivy — APPROVED** [review]
LGTM, really nice.

File: Examples/CppInPlace/Documentation.rst (hunk 1 above)
Comment (blowekamp) [reply Examples/CppInPlace/Documentation.rst]: done

File: Examples/CppInPlace/Documentation.rst [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+.. code-block:: c++
+
+  Image UnaryMinus(Image &&image1);
+  Image SmoothingRecursiveGaussian(Image &&image1, std::vector<double> sigma, bool normalizeAcrossScale);
+
+Additionally, most C++ overloaded Image numeric operators such as `+`, `-`, `*`, `%`, etc. support inplace execution.
+
+The SimpleITK Image class supports move semantics with the constructor and assignment operator. Since the
+Image class also uses a lazy copy on write implement for the constructors and assignment there is not a  significant
```
Comment (dave3d) [inline Examples/CppInPlace/Documentation.rst:32]: copy-on-write

File: Examples/CppInPlace/Documentation.rst [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+direct performance benefit using these operations. However, the move semantics can be useful to reduce the number
+of Image objects referencing the same image buffer to prevent unnecessary copies. Lastly, if a rvalue Image object is
+passed to a filter and Image object's buffer is not unique, then the image buffer will be copied before the filter is
```
Comment (dave3d) [inline Examples/CppInPlace/Documentation.rst:35]: 'that Image object's buffer' maybe?

**Review by dave3d — APPROVED** [review]
I suggest using 'in-place', since inplace isn't a word

**Pull request merged**